hummingbot/connector/exchange/bitmax/bitmax_exchange.py

```diff
@@ -0,0 +1,961 @@
+            self.trigger_event(MarketEvent.OrderCancelled,
+                               OrderCancelledEvent(
+                                   self.current_timestamp,
+                                   client_order_id,
+                                   exchange_order_id))
+            tracked_order.cancelled_event.set()
+            self.stop_tracking_order(client_order_id)
+        elif tracked_order.is_failure:
+            self.logger().info(f"The market order {client_order_id} has failed according to order status API. "
+                               f"Reason: {bitmax_utils.get_api_reason(order_msg['reason'])}")
+            self.trigger_event(MarketEvent.OrderFailure,
+                               MarketOrderFailureEvent(
+                                   self.current_timestamp,
+                                   client_order_id,
+                                   tracked_order.order_type
+                               ))
+            self.stop_tracking_order(client_order_id)
+        elif tracked_order.is_done:
+            price = Decimal(order_msg["ap"])
+            tracked_order.executed_amount_base = Decimal(order_msg["cfq"])
+            tracked_order.executed_amount_quote = price * tracked_order.executed_amount_base
+            tracked_order.fee_paid = Decimal(order_msg["cf"])
+            tracked_order.fee_asset = order_msg["fa"]
+
+            self.trigger_event(
+                MarketEvent.OrderFilled,
+                OrderFilledEvent(
+                    self.current_timestamp,
+                    client_order_id,
+                    tracked_order.trading_pair,
+                    tracked_order.trade_type,
+                    tracked_order.order_type,
+                    price,
+                    tracked_order.executed_amount_base,
+                    TradeFee(0.0, [(tracked_order.fee_asset, tracked_order.fee_paid)]),
+                    exchange_order_id
+                )
+            )
+            event_tag = MarketEvent.BuyOrderCompleted if tracked_order.trade_type is TradeType.BUY else MarketEvent.SellOrderCompleted
+            event_class = BuyOrderCompletedEvent if tracked_order.trade_type is TradeType.BUY else SellOrderCompletedEvent
+            self.trigger_event(
+                event_tag,
+                event_class(
+                    self.current_timestamp,
+                    client_order_id,
+                    tracked_order.base_asset,
+                    tracked_order.quote_asset,
+                    tracked_order.fee_asset,
+                    tracked_order.executed_amount_base,
+                    tracked_order.executed_amount_quote,
+                    tracked_order.fee_paid,
+                    tracked_order.order_type,
+                    exchange_order_id
+                )
+            )
+            self.stop_tracking_order(client_order_id)
+
+    async def _process_trade_message(self, trade_msg: Dict[str, Any]):
```

Comment:
This does not seem to be called anywhere. This method should be called either in the `_user_stream_event_listener` method or the `_update_order_status`

Reply:
Marking this as `resolved` and we can track it over at https://github.com/CoinAlpha/hummingbot/pull/2823#discussion_r564759701